src/otx/v2/adapters/torch/mmengine/mmseg/dataset.py

```diff
@@ -0,0 +1,252 @@
+        # in runner.build_dataloader?
+        seed = kwargs.get("seed", 0)
+
+        # build sampler
+        if sampler is None:
+            sampler = DATA_SAMPLERS.build(
+                {"type": "DefaultSampler", "shuffle": shuffle},
+                default_args={"dataset": dataset, "seed": seed},
+            )
+
+        init_fn = partial(worker_init_fn, num_workers=num_workers, rank=rank, seed=seed) if seed is not None else None
+        if digit_version(torch.__version__) >= digit_version("1.8.0"):
+            kwargs["persistent_workers"] = persistent_workers
+
+        dataloader = TorchDataLoader(
+            dataset,
+            batch_size=batch_size,
+            sampler=sampler,
+            num_workers=num_workers,
+            collate_fn=partial(pseudo_collate),
+            pin_memory=pin_memory,
+            worker_init_fn=init_fn,
+            drop_last=drop_last,
+            **kwargs,
+        )
+        sampler_cfg = sampler if isinstance(sampler, dict) else {"type": f"{sampler.__class__.__qualname__}"}
+        dataset_cfg = dataset.configs if hasattr(dataset, "configs") else dataset
+        dataloader.configs = {
+            "batch_size": batch_size,
+            "sampler": sampler_cfg,
+            "num_workers": num_workers,
+            "collate_fn": {"type": "pseudo_collate"},
+            "pin_memory": pin_memory,
+            "shuffle": shuffle,
+            "dataset": dataset_cfg,
+        }
+        return dataloader
+
+    def _build_dataset(
+        self,
+        subset: str,
+        pipeline: list | None = None,
+        config: dict | None = None,
+    ) -> TorchDataset | None:
+        """Builds a TorchDataset object for the given subset using the specified pipeline and configuration.
+
+        Args:
+            subset (str): The subset to build the dataset for.
+            pipeline (list | None, optional): The pipeline to use for the dataset.
+                Defaults to None.
+            config (dict | None, optional): The configuration to use for the dataset.
+                Defaults to None.
+
+        Returns:
+            Optional[TorchDataset]: The built TorchDataset object, or None if the dataset is empty.
+        """
+        def check_and_convert_to_tuple(pipeline: list[dict] | None) -> list[dict] | None:
+            """Check if the pipeline is None, and if not, convert any lists in the pipeline to tuples.
+
+            Args:
+                pipeline (list[dict] | None): The pipeline to check and convert.
+
+            Returns:
+                list[dict] | None: The converted pipeline, or None if the input was None.
+            """
+            if pipeline is None:
+                return None
+            for step in pipeline:
+                for key, value in step.items():
+                    if isinstance(value, list):
+                        step[key] = tuple(value)
+            return pipeline
```

Comment:
When `config.yaml` gets dumped into the workspace, `OmegaConf` turns Python tuples into YAML list style - _the issue is actually not tuple-specific; the same problem occurs with any other type that is not supported by OmegaConf_. Because tuples were converted to lists during the dump, upon loading, they're treated as lists of values. Essentially, here's where we dump config.yaml: https://github.com/openvinotoolkit/training_extensions/blob/feature/v2/src/otx/v2/cli/utils/workspace.py#L91

We have a couple of options to fix this. One is to convert the list back to a tuple on our end. Alternatively, we can explore a workaround like the one mentioned here: https://github.com/omry/omegaconf/issues/392. However, based on the attached issue, it's not the most ideal solution IMO.

Reply:
I understand, we should apply global solution for this in next phase. In detection case, the only problem is LoadResizeDataFromOTXDataset pipeline, so I just fix like this
![image](https://github.com/openvinotoolkit/training_extensions/assets/106788349/b4a7b469-d92d-41d5-b9b0-2d4f493d9eb3)